wayback/_client.py

```diff
@@ -400,17 +400,21 @@ def send(self, *args, **kwargs):
         retries = 0
         while True:
             try:
+                logger.debug("sending %s", kwargs)
                 result = super().send(*args, **kwargs)
                 if retries >= maximum or not self.should_retry(result):
                     if result.status_code == 429:
+                        logger.warning("caught rate limit error: %s", result.content)
```

Comment:
I don’t think we really need to log here, or at least this shouldn’t be a `warning`. If the user (of this library) gets an exception (which they will because we `raise` here), they don’t need a warning logged as well.

Reply:
I believe 429 errors are raised and then caught. I think it's useful to know when it's happening?

Reply:
Where does it get caught?

Reply:
OK, so I tested this in practice just to make sure, and the exception we raise at this location does not get caught.

I think the confusion here is that the exception handler here is only catching networking errors, not non-200 responses, and what you were really looking for is a warning that Wayback returned a 429 and we are going to retry. That needs to happen with an `else` on the `if` before this one (`if retries >= maximum or not self.should_retry(result):`). I’m going to go ahead and add that here.

Also, in practice, it seems like the CDX server is not sending a `Retry-After` header to tell us how long to pause for (so it doesn’t match [what I remember](https://github.com/edgi-govdata-archiving/wayback/issues/137#issuecomment-1790980712), but it could be the memento server responds differently and I was thinking of that 🤷 ), but it *does* seem like we could stand to pause a little longer when we get a 429, and that would be a good change to make here, too. I’ll see about doing that as a separate PR so this stays focused on logging.

Reply:
Thanks for investigating further @Mr0grog! Is there anything we should communicate to Internet Archive?

Reply:
If you’re still talking with them about it, maybe let them know it would be nice if they sent a `Retry-After` header with their 429 responses, but otherwise, no. This is really all just issues on our side.

Reply:
Anyway, I’ll try and get this merged and released later today.